src/blockmap.jl

```diff
@@ -57,9 +69,7 @@ function Base.hcat(As::Union{LinearMap,UniformScaling}...)
     end
     nrows == 0 && throw(ArgumentError("hcat of only UniformScaling-like objects cannot determine the linear map size"))
```

Comment:
I still feel uncomfortable that this changes Base behaviour, also, the error message is somewhat different. One way out would be to call this function `_hcat` and then define
```julia
for k = 1:8 # is 8 sufficient?
      Is = ntuple(n->:($(Symbol(:A,n))::UniformScaling), Val(k-1))
      L = :($(Symbol(:A,k))::LinearMap)
      args = ntuple(n->Symbol(:A,n), Val(k))

      @eval Base.hcat($(Is...), $L, As::Union{LinearMap,UniformScaling}...) = _hcat($(args...), As...)
      @eval Base.vcat($(Is...), $L, As::Union{LinearMap,UniformScaling}...) = _vcat($(args...), As...)
      @eval Base.hvcat(rows::Dims, $(Is...), $L, As::Union{LinearMap,UniformScaling}...) = _hvcat(rows, $(args...), As...)
end
```

Reply:
The change in the error message was a left-over from earlier attempts (and `vcat` had the `hcat` message). As a quick fix, I plugged in the same error message as in Base. As an intermediate step, I'll make a commit with your other suggestions and this quick fix. If I find the time, I'll try to use your above code, but if anything fails, I'll leave it at that for the moment.

Reply:
Copy, paste, and it works... not my code, obviously. 🤣